### PR TITLE
JAVA-2053:

### PR DESCRIPTION
**Pull request by jyemin**
When no stream factory factory is provided, ensure that any EventLoopGroup instances are shutdown cleanly.

### CODE REVIEW

File: driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java [hunk 1]
```diff
@@ -95,7 +93,7 @@ public static Builder builder(final MongoClientSettings settings) {
         private WriteConcern writeConcern = WriteConcern.ACKNOWLEDGED;
         private ReadConcern readConcern = ReadConcern.DEFAULT;
         private CodecRegistry codecRegistry = MongoClients.getDefaultCodecRegistry();
-        private StreamFactoryFactory streamFactoryFactory = createDefaultStreamFactoryFactory();
+        private StreamFactoryFactory streamFactoryFactory;
```
Comment (jyemin) [inline driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java]: A slight change in behavior of this class to not have a default stream factory factory, but it's necessary to not unnecessarily leak an NioEventLoopGroup.

File: driver-async/src/main/com/mongodb/async/client/MongoClientImpl.java [hunk 2]
```diff
@@ -40,15 +43,22 @@
     private final Cluster cluster;
     private final MongoClientSettings settings;
     private final AsyncOperationExecutor executor;
+    private final Closeable externalResourceCloser;
 
-    MongoClientImpl(final MongoClientSettings settings, final Cluster cluster) {
-        this(settings, cluster, createOperationExecutor(settings, cluster));
+    MongoClientImpl(final MongoClientSettings settings, final Cluster cluster, final Closeable externalResourceCloser) {
```
Comment (jyemin) [inline driver-async/src/main/com/mongodb/async/client/MongoClientImpl.java]: This isn't pretty but at least it's not public.

Comment (rozza) [inline driver-async/src/main/com/mongodb/async/client/MongoClientImpl.java]: Acknowledged

File: driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java (hunk 1 above)
Comment (rozza) [inline driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java]: Ack

**Pull request closed without merging**